### PR TITLE
Implement browser extension Sourcegraph URL dropdown for ease of URL switching

### PR DESCRIPTION
**Pull request by erzhtor**
Closes https://github.com/sourcegraph/sourcegraph/issues/29030.
Closes https://github.com/sourcegraph/sourcegraph/issues/25778.

#### Description

This PR:
- Implement browser extension Sourcegraph URL dropdown for ease of URL switching
- Updates options page UI

#### How to test

1. `sg run bext`
2. Open browser extension options page
3. Check that the input field suggests **all previously successfully validated URLs**

#### Screenshots
| Before | After |
| -: | :- |
|image | image |
| n/a | image |
| image | image |
|image | image |

#### Before merging

- [ ] Test on different code hosts (if applicable)
    - [ ] GitHub
    - [ ] Gitlab
    - [ ] GitHub Enterprise
    - [ ] Refined GitHub
    - [ ] Phabricator
    - [ ] Phabricator integration
    - [ ] Bitbucket
    - [ ] Bitbucket integration

- [ ] Test on different browsers (if applicable)
    - [ ] Chrome
    - [ ] Firefox
    - [ ] Safari
- [x] Add change log message to [client/browser/CHANGELOG.md](./client/browser/CHANGELOG.md), under "Unreleased" section. (if applicable)

### CODE REVIEW

**Comment by jjinnii** [comment]
@erzhtor Looking really good! Agree with Beatrix's slack comment that removing the code hosts makes the extension look really clean :) Just a few items I'd like to check: 

- My bad for not including in the designs, but I think we need to fine-tune the advanced settings section to align with the new bext UI. Please take a look at this [wireframe](https://www.figma.com/file/anIrmiCUE4lm7Fib89RjU2/RFC-459-Browser-extension-fallback-to-Sourcegraph.com?node-id=1737%3A4852) that I just created here for updates. I removed "Configuration", as there's no need for a double header, and reduced the font size. 
- The container for bext is wider than the designs (416 px versus 368 px), leading to differences in layout (the alignment of the Sourcegraph Cloud and Documentation buttons for example). Could we adjust the width + height of bext to match the mockups? I think the design looks tighter with the reduced width (and blocking less screenspace is always a plus with extensions) 
- The dropdown is a bit hard to distinguish from the background. Is it possible to [add the shadow](https://www.figma.com/file/anIrmiCUE4lm7Fib89RjU2/RFC-459-Browser-extension-fallback-to-Sourcegraph.com?node-id=1622%3A6472) to the dropdown? (for both light and dark mode). 
- For dark mode specifically, can we change the background color of the dropdown to match the bg color of the input field? That will help further to visually distinguish the dropdown.  
Screen Shot 2022-01-07 at 11 49 14 AM

- Is there another issue for exploring/adding the [metric ](https://www.figma.com/file/anIrmiCUE4lm7Fib89RjU2/RFC-459-Browser-extension-fallback-to-Sourcegraph.com?node-id=1622%3A7203)for # of users with >1 URL stored in bext local storage?

Let me know if you'd prefer another issue for these updates!

**Comment by jjinnii** [comment]
@erzhtor  It's looking so good! Thank you for making those updates so quickly :) Just one super small thing - Could you double-check the spacing below "Experimental link previews" when the Settings is expanded? It looks a bit large -- the spacing is 3 rem to match the space from the border to "Sourcegraph URL". 

Screen Shot 2022-01-11 at 11 39 41 AM

Other than that, looks great and no other comments on the UI from me! Let me know if you need any help with the metric.

**Review by jjinnii — APPROVED** [review]

**Comment by jjinnii** [comment]
Actually @erzhtor quick question. Right now, I'm seeing sourcegraph.com in the dropdown when the only instance I've successfully connected is Sourcegraph Cloud.

Screen Shot 2022-01-11 at 12 56 35 PM

If the user has only ever configured Sourcegraph Cloud, is it possible for us to not show the dropdown? Until a user has successfully connected a second instance, I think it will confuse users to show a dropdown with the singular Sourcegraph Cloud option.

**Comment by erzhtor** [comment]
@jjinnii, thanks for the review! Updated for spacing & showing dropdown for 2+ URLs.

File: client/browser/src/shared/tracking/eventLogger.tsx [hunk 1]
```diff
@@ -26,35 +25,37 @@ const uidKey = 'sourcegraphAnonymousUid'
...
+                this.innerTelemetryService.log(eventName, eventProperties).catch(console.error)
+            }
+        })
     }
     public logViewEvent(eventName: string, eventProperties?: any): void {
-        if (this.isEnabled) {
-            this.innerTelemetryService.logViewEvent(eventName, eventProperties)
-        }
+        // Wait for this.isEnabled to get a new value
+        setTimeout(() => {
```
Comment (tjkandala) [inline client/browser/src/shared/tracking/eventLogger.tsx:50]: In what case is this necessary?

**Review by tjkandala — APPROVED** [review]
LGTM, nice feature!

**Comment by jjinnii** [comment]
@erzhtor Looks great, and thanks again for making those updates so quickly 💯 Just responding to[ your question in Slack](https://sourcegraph.slack.com/archives/C01LZKLRF0C/p1642007752067800?thread_ts=1641996050.062200&cid=C01LZKLRF0C) about the metrics: If I'm understanding correctly the new event, Bext_NumberURLs can measure:
- Number of times a users switch between instances. (although this could be a handful of users switching multiple times daily skewing numbers)
- Flat # of users who have >1 URL stored in the next dropdown. (< this would be most helpful for us) 

If that's correct, then yes that was what I had in mind! Thank you for working through the security implications and getting that set up.

**Review by attfarhan — APPROVED** [review]

File: client/browser/src/shared/tracking/eventLogger.tsx (hunk 1 above)
Comment (erzhtor) [reply client/browser/src/shared/tracking/eventLogger.tsx:50]: When the EventLogger is created and `.log` method is immediately called, the `isEnabled` doesn't get updated.

**Comment by erzhtor** [comment]
@jjinnii, thanks for the reply!

> * Number of times users switch between instances. (although this could be a handful of users switching multiple times daily skewing numbers)

Yeah, the event will be called every time the user **switches back** to sourcegraph.com from the self-hosted instance because we can only log to the cloud when connected to it.

> * Flat # of users who have >1 URL stored in the next dropdown. (< this would be most helpful for us)

I think we calculate amplitude using grouping by `anonUserId/unique. Or do you also want to track # URLs each user has available in the dropdown?

**Comment by jjinnii** [comment]
> @jjinnii, thanks for the reply!
> 
> > * Number of times users switch between instances. (although this could be a handful of users switching multiple times daily skewing numbers)
> 
> Yeah, the event will be called every time the user **switches back** to sourcegraph.com from the self-hosted instance because we can only log to the cloud when connected to it.
> 
> > * Flat # of users who have >1 URL stored in the next dropdown. (< this would be most helpful for us)
> 
> I think we calculate amplitude using grouping by `anonUserId/unique. Or do you also want to track # URLs each user has available in the dropdown?

Nope, just the number of users is sufficient! I don't think tracking the outlier # of users who have >2/3 is all that important.

**Pull request merged**